src/server/controller/plugin.ts

```diff
@@ -0,0 +1,37 @@
+import { PostgresGateway } from "../gateways";
+import Gateway from "../gateways/gateway";
+import { Trade } from "../types";
+
+/* To-do: We'll need to further abstract Gateway. */
+export default class Controller extends Gateway {
```

Comment:
**Annotation:** Controller is not meant to be a `Gateway`. It just so happens that the interface is useful enough to also be used here, and looking ahead, conceivably in services going forward. We will have to abstract `Gateway` into even higher scope.